source/common/http/http2/codec_impl.cc

```diff
@@ -125,6 +131,48 @@ void ConnectionImpl::StreamImpl::encodeTrailers(const HeaderMap& trailers) {
     parent_.sendPendingFrames();
   }
 }
+void ConnectionImpl::StreamImpl::readDisable(bool disable) {
+  ENVOY_CONN_LOG(debug, "Stream {} disabled: disable {}, unconsumed_bytes {}", parent_.connection_,
+                 stream_id_, disable, unconsumed_bytes_);
+  if (disable) {
+    ++read_disable_count_;
+  } else {
+    --read_disable_count_;
+    if (!buffers_overrun()) {
+      nghttp2_session_consume(parent_.session_, stream_id_, unconsumed_bytes_);
+      unconsumed_bytes_ = 0;
+      parent_.sendPendingFrames();
+    }
+  }
+}
+
+void ConnectionImpl::StreamImpl::pendingRecvBufferHighWatermark() {
+  ENVOY_CONN_LOG(debug, "recv buffer over limit ", parent_.connection_);
+  ASSERT(pending_receive_buffer_high_watermark_called_ == false);
```

Comment:
I think for booleans we don't ned equality with the value, just something like `!pending_receive..`.